src/main/java/org/reaktivity/nukleus/kafka/internal/stream/KafkaCacheServerProduceFactory.java

```diff
@@ -1001,11 +1002,19 @@ private void doProduceInitialData(
                     final int initialPadding = fan.initialPadding;
                     final int initialBudget = fan.initialBudget;
                     final int reserved = Math.min(remaining + initialPadding, initialBudget);
+                    final int dirtyFlag = nextEntry.flags();
 
                     assert partitionOffset >= cursor.offset : String.format("%d >= %d", partitionOffset, cursor.offset);
 
+                    init:
                     if (reserved >= initialPadding)
                     {
+                        if (dirtyFlag != CACHE_ENTRY_FLAGS_COMPLETED)
+                        {
+                            cursor.advance(partitionOffset + 1);
+                            break init;
```

Comment:
We need `doFlushServerReply(traceId);` before the `break init;` to fully mimic successful produce case below when entry is marked dirty, agree?